ci/infra/testrunner/tests/test_istio_deployment.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import pytest
+import requests
+import time
+
+
+GATEWAY_HTTPBIN = ("""
+---
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: httpbin-gateway
+  namespace: default
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - hosts:
+    - '*'
+    port:
+      name: http
+      number: 80
+      protocol: HTTP
+EOF
+""")
+
+VIRTUALSERVICE_HTTPBIN = ("""
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: httpbin
+spec:
+  hosts:
+  - '*'
+  gateways:
+  - httpbin-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /status
+    - uri:
+        prefix: /delay
+    route:
+    - destination:
+        port:
+          number: 8000
+        host: httpbin
+EOF
+""")
+
+
+@pytest.mark.pr
+def test_istio_deployment(deployment, platform, skuba, kubectl):
+    logger = logging.getLogger("testrunner")
+    logger.info("Deploy Istio")
+    istioctl = ("""
+                istioctl --kubeconfig={config} manifest apply \
+                        --set profile=default \
+                        --set addonComponents.prometheus.enabled=false \
+                        --set hub=registry.suse.de/devel/caasp/5/containers/containers/caasp/v5 \
+                        --set tag=1.5.4 \
+                        --set values.pilot.image=istio-pilot \
+                        --set values.global.proxy.image=istio-proxyv2 \
+                        --set values.global.proxy_init.image=istio-proxyv2
+                 """.format(config=kubectl.get_kubeconfig()))
+
+    kubectl.utils.runshellcommand(istioctl)
+
+    kubectl.run_kubectl("-n istio-system wait --for=condition=available deploy/istio-ingressgateway --timeout=3m")
+
+    logger.info("Deploy httpbin sample app")
+    kubectl.run_kubectl("create -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/httpbin/httpbin.yaml")
+
+    logger.info("Create the istio config")
+    kubectl.run_kubectl("apply -f - << EOF " + GATEWAY_HTTPBIN)
+    kubectl.run_kubectl("apply -f - << EOF " + VIRTUALSERVICE_HTTPBIN)
+
+    # Wait for istio to digest the config
+    time.sleep(100)
+
+    nodePort = kubectl.run_kubectl("-n istio-system get service/istio-ingressgateway -o jsonpath='{ .spec.ports[1].nodePort }'")
+
+    wrk_idx = 0
+    ip_addresses = platform.get_nodes_ipaddrs("worker")
+
+    url = "{protocol}://{ip}:{port}{path}".format(protocol="http", ip=str(ip_addresses[wrk_idx]), port=str(nodePort), path="/status/200")
+    r = requests.get(url, headers={'host': 'httpbin.example.com'})
+
+    assert 200 == r.status_code
+
+    # Cleanup
```

Comment:
As per the previous comment, this could be the cleanup function for the istio fixture

Reply:
you mean better create a separate function which cleans up everything?

Reply:
No I mean you can follow this patter for a fixture:
```
@pytest.fixture
def istio(request, kubectl):
    def cleanup():
        # cleanup logic here

    request.addfinalizer(cleanup)

    # istio delployment logic here
```

Reply:
as discussed in RC, we'll use the fixture pattern in the future

Reply:
For the record, the reason is that there will be more than one test in the same module, which requires defining the fixture `istio` with scope 'module', but this conflicts with the scope 'function' the fixtures it requires, such as `kubeclt`.